actionpack/lib/abstract_controller/base.rb

```diff
@@ -148,7 +148,9 @@ def process(action, *args)
 
       @_response_body = nil
 
-      process_action(action_name, *args)
+      ActiveSupport::ExecutionContext.set(controller: self) do
```

Comment:
I just realized that there might be a subtle difference between Query Logs and Error Reporting.

In the error reporting case, the exception will bubble up to the executor, so once it's handled this `controller` metadata would have been removed.

So either combining both doesn't make sense, or either Action Controller and Active Job should not unset their metadata.

Reply:
I changed it to not use `.set` as I think it makes sense anyway.